routes/courseRoutes.js

```diff
@@ -46,15 +47,56 @@ router.get('/creator/:id', requireLogin, async (req, res) => {
 
 });
 
-//Get all courses
-router.get('/', async (req, res) => {
 
+const topFeedbackOptionForCourses = async (courses) => {
+	// Fetch all feedback options in a single query
+	const feedbackOptions = await FeedbackOptionsModel.find().lean();
+
+	// Create a map of feedback options by ID for quick lookup
+	const feedbackOptionsMap = feedbackOptions.reduce((map, option) => {
+		map[option._id] = option;
+		return map;
+	}, {});
+
+	// Process each course
+	courses.forEach(course => {
+		if (course.feedbackOptions && course.feedbackOptions.length > 0) {
```

Comment:
Minor: Isn't this redundant? If course.feedbackOptions is an array of length 0, the second condition will never be reached, as an empty array is falsy.

Reply:
Technically we're only interested in the second condition. We just use the first condition to short circuit the logic on courses where feedback options don't exist since it is a new field in the model.  
[] passes the first condition so we know that we _can_ check the second. In an ideal world where every course is updated with the new model we only check the second condition